client/src/KeyguardClient.ts

```diff
@@ -10,21 +14,56 @@ export class KeyguardClient {
         : window.location.origin === 'https://accounts.nimiq-testnet.com' ? 'https://keyguard-next.nimiq-testnet.com'
         : `${location.protocol}//${location.hostname}:8000/src`;
 
+    private static parseId(id: string) {
+        if (id.substr(0, 1) !== 'K') {
+            throw new Error('keyId must start with K');
+        }
+        const parsedId = parseInt(id.substr(1), 10);
+        if (isNaN(parsedId)) {
+            throw new Error('keyId cannot be parsed');
+        }
+        return parsedId;
+    }
+
+    private static publicToInternal<T extends Request>(object: any)
+        : PublicToInternal<T> {
+        if (object && object.keyId) {
+            object.keyId = KeyguardClient.parseId(object.keyId);
+        }
+        if (object && object.id) {
+            object.id = KeyguardClient.parseId(object.id);
+        }
+        return object;
+    }
+
+    // Not really well typed. Return type gets any. Feel free to improve.
+    private static internalToPublic(result: any): InternalToPublic<typeof result> {
```

Comment:
Precisely, I get back unknown from keyguard (like the keyguard gets a request of type unknown). It would be cleaner to use a response parser.
The result type gets, for redirect requests at least, lost anyway in the Observable.fire() currently. :/
I already started working on another PR which hopefully will solve some of those issues, but I felt there are still more important things to do.